### PR TITLE
refactor: remove execution_chain GenServer

### LINKED ISSUE #1193: Remove ExecutionChain genserver
We need to replace all `GenServer.casts` with function calls by storing the `ExecutionChain`'s state in the db.

### PR DESCRIPTION
**Pull request by avilagaston9**
- Stores the `ExecutionChain`'s state in the db.
- Replaces all `GenServer.casts` with function calls.

closes #1193

### CODE REVIEW

File: lib/lambda_ethereum_consensus/execution/execution_chain.ex [hunk 1]
```diff
@@ -266,4 +269,15 @@ defmodule LambdaEthereumConsensus.Execution.ExecutionChain do
 
   defp slots_per_eth1_voting_period(),
     do: ChainSpec.get("EPOCHS_PER_ETH1_VOTING_PERIOD") * ChainSpec.get("SLOTS_PER_EPOCH")
+
+  defp persist_execution_state(state) do
+    put("", state)
+  end
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/execution/execution_chain.ex:275]: nit, maybe a one-liner here like the get?

File: lib/lambda_ethereum_consensus/execution/execution_chain.ex [hunk 2]
```diff
@@ -266,4 +269,15 @@ defmodule LambdaEthereumConsensus.Execution.ExecutionChain do
...
+
+  defp fetch_execution_state(), do: get("")
+
+  defp fetch_execution_state!() do
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/execution/execution_chain.ex:279]: Can we add a spec to the get and persist functions so it gives a clear indication of what shape the "execution_state" has?

**Review by Arkenan — APPROVED** [review]
LGTM with small comments

File: lib/lambda_ethereum_consensus/execution/execution_chain.ex (hunk 1 above)
Comment (avilagaston9) [reply lib/lambda_ethereum_consensus/execution/execution_chain.ex:275]: Done!

File: lib/lambda_ethereum_consensus/execution/execution_chain.ex (hunk 2 above)
Comment (avilagaston9) [reply lib/lambda_ethereum_consensus/execution/execution_chain.ex:288]: Good idea!

**Pull request merged**